src/main/java/org/maxgamer/quickshop/shop/ContainerShop.java

```diff
@@ -747,9 +816,16 @@ public void refresh() {
         }
         if (displayItem != null) {
             displayItem.remove();
+        }
+
+        checkDisplay();
+
+        if (!isLeftShop) {
             initDisplayItem();
             displayItem.spawn();
+            Util.debugLog("Not left shop, respawning item!");
         }
```

Comment:
Recommend to remove this debug log, since not a left shop case is the most cases